webapp/src/components/chat/persona/PromptEditor.tsx

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import {
+    Button,
+    Popover,
+    PopoverSurface,
+    PopoverTrigger,
+    Textarea,
+    makeStyles,
+    shorthands,
+    tokens
+} from '@fluentui/react-components';
+import React from 'react';
+import { AlertType } from '../../../libs/models/AlertType';
+import { useAppDispatch } from '../../../redux/app/hooks';
+import { addAlert } from '../../../redux/features/app/appSlice';
+import { Info16 } from '../../shared/BundledIcons';
+
+const useClasses = makeStyles({
+    root: {
+        display: 'flex',
+        flexDirection: 'column',
+        ...shorthands.gap(tokens.spacingVerticalSNudge),
+    },
+    horizontal: {
+        display: 'flex',
+        ...shorthands.gap(tokens.spacingVerticalSNudge),
+        alignItems: 'center',
+    },
+    controls: {
+        display: 'flex',
+        marginLeft: 'auto',
+    },
+});
+
+interface PromptEditorProps {
+    title: string;
+    prompt: string;
+    isEditable: boolean;
+    info: string;
+    modificationHandler?: (value: string) => Promise<void>;
+}
+
+export const PromptEditor: React.FC<PromptEditorProps> = ({
+    title,
+    prompt,
+    isEditable,
+    info,
+    modificationHandler,
+}) => {
+    const classes = useClasses();
+    const dispatch = useAppDispatch();
+    const [value, setValue] = React.useState<string>(prompt);
+
+    React.useEffect(() => {
+        setValue(prompt);
+    }, [prompt]);
```

Comment:
does the `prompt` change outside of this component while its being rendered?

Reply:
Yes likely, prompt is passed in.